src/protocol/vm/tycho_decoder.rs

```diff
@@ -113,15 +117,15 @@ impl TryFromWithBlock<ComponentWithState> for VMPoolState<PreCachedDB> {
             .map(H160::from_bytes)
             .collect();
 
-        let adapter_file_path = format!(
-            "src/protocol/vm/assets/{}",
-            to_adapter_file_name(
-                snapshot
-                    .component
-                    .protocol_system
-                    .as_str(),
-            )
-        );
+        let protocol_name = snapshot
+            .component
+            .protocol_system
+            .strip_prefix("vm:")
+            .unwrap_or({
+                Err(InvalidSnapshotError::VMError(SimulationError::DecodingError("Invalid protocol system name: VMPoolStates expect a protocol name starting with 'vm:'".to_string())))?
```

Comment:
What err inception is going on here? Did you just look for the closest DecodingError?

Reply:
`InvalidSnapshotError` only has 'MissingAttribute', 'VMError' and 'ValueError'. Initially felt VMError was the most appropriate, but a simple ValueError is probably better?